nsxt/policy_utils.go

```diff
@@ -421,21 +448,47 @@ func nsxtPolicyPathResourceImporter(d *schema.ResourceData, m interface{}) ([]*s
 	return rd, nil
 }
 
-// This importer function accepts policy path only as import ID
-func nsxtPolicyPathOnlyResourceImporter(d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
-	return nsxtPolicyPathResourceImporterHelper(d, m)
+func getMultitenancyPathExample(pathExample string) string {
+	return fmt.Sprintf("%s or /orgs/[org]/projects/[project]%s", pathExample, pathExample)
 }
 
-func nsxtVPCPathResourceImporter(d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
-	rd, err := nsxtPolicyPathResourceImporterHelper(d, m)
-	if err != nil {
-		return rd, err
+// This importer function accepts policy path or resource ID
+func getFriendlyPolicyPathOrIDResourceImporter(pathExample string) func(d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+	return func(d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+		rd, err := nsxtPolicyPathResourceImporterHelper(d, m)
+		if errors.Is(err, ErrUnexpectedPolicyPath) {
+			return rd, fmt.Errorf("Invalid policy path %s; expected format: %s", d.Id(), pathExample)
+		}
+		if errors.Is(err, ErrNotAPolicyPath) {
+			if !isValidResourceID(d.Id()) {
+				return rd, fmt.Errorf("Invalid policy path or id %s; expected path format: %s", d.Id(), pathExample)
```

Comment:
Do you think this error message makes sense in the case we fail to validate the resource id? The user did not specify a path.

Reply:
There is little sense to give an example of id.. but I rephrased it to avoid confusion, does it look better now?